### PR TITLE
Add weather API aggregator service

### PR DESCRIPTION
**Pull request by szurgot207604**

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Tests?

File: src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public class WeatherApiAggregatorService implements WeatherApiInterface {
+
+    private List<WeatherApiInterface> services;
+
+    public WeatherApiAggregatorService(List<WeatherApiInterface> services) {
+        this.services = services;
+    }
+
+    @Override
+    public WeatherApiData getData(Double latitude, Double longitude) {
```
Comment (ghost) [inline src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java:19]: Test it

File: src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        }
+
+        if (servicesCounter == 0) {
+            return null;
+        }
+
+        return this.averageData(aggregatedData, servicesCounter);
+    }
+
+    private WeatherApiData sumData(WeatherApiData data1, WeatherApiData data2) {
```
Comment (ghost) [inline src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java:48]: Test it

File: src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                data1.getTemperatureMax() + data2.getTemperatureMax(),
+                data1.getPressure() + data2.getPressure(),
+                data1.getHumidity() + data2.getHumidity(),
+                data1.getWindSpeed() + data2.getWindSpeed(),
+                data1.getWindDirection() + data2.getWindDirection(),
+                data1.getCloudsDensity() + data2.getCloudsDensity()
+        );
+    }
+
+    private WeatherApiData averageData(WeatherApiData data, Integer divider) {
```
Comment (ghost) [inline src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java:60]: Test it

**Review by ghost — CHANGES_REQUESTED** [review]
Marked methods that need tests

File: src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java (hunk 2 above)
Comment (szurgot207604) [reply src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java:48]: won't test private method

File: src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java (hunk 3 above)
Comment (szurgot207604) [reply src/main/java/pl/lodz/p/ics/weatherapp/api/services/WeatherApiAggregatorService.java:60]: won't test private method

**Review by peczek207585 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**